app/writer.py

```diff
@@ -65,6 +69,9 @@ def set_times(e, times):
     e.add('dtstart', times[0])
     e.add('dtend', times[1])
 
+def remove_events_on_menu(cal, menu_start):
+    """ Removes any events from `cal` that have been "rescraped" """
```

Comment:
It just replaces the list of subcomponents with only the ones that occur before the first event on the current menu.  `list[:]` just replaces the list in place.